header/sync/sync_head.go

```diff
@@ -98,13 +98,14 @@ func (s *Syncer) networkHead(ctx context.Context) (*header.ExtendedHeader, error
 func (s *Syncer) incomingNetHead(ctx context.Context, netHead *header.ExtendedHeader) pubsub.ValidationResult {
 	// Try to short-circuit netHead with append. If not adjacent/from future - try it as new network header
 	_, err := s.store.Append(ctx, netHead)
-	switch err {
-	case nil:
+	if err == nil {
 		// a happy case where we appended maybe head directly, so accept
 		return pubsub.ValidationAccept
-	case header.ErrNonAdjacent:
+	}
+	var nonAdj *header.ErrNonAdjacent
+	if errors.As(err, &nonAdj) {
 		// not adjacent, maybe we've missed a few headers or its from the past
-	default:
+	} else {
```

Comment:
Wonder if there is a way to shorten/avoid `nonAdj` in case we don't need it's body. 

We can add a debug log here as well

Reply:
Let's just leave it as is -- added debug log.